exams/management/commands/simulate_sftp_responses.py

```diff
@@ -0,0 +1,158 @@
+
+                    if not keep:
+                        sftp_conn.remove(path)
+
+    def handle_ead(self, sftp_conn, remote_path, ratio):
+        """Handle an EAC file"""
+        now = datetime.now(pytz.utc)
+        result_file = io.StringIO()
+        writer = csv.DictWriter(result_file, [
+            'ClientAuthorizationID',
+            'ClientCandidateID',
+            'Status',
+            'Date',
+            'Message',
+        ], delimiter='\t')
+        writer.writeheader()
+        with sftp_conn.open(remote_path, mode='r') as eac_file:
+            for row in csv.DictReader(eac_file, delimiter='\t'):
+                cid = row['ClientCandidateID']
+                aid = row['ClientAuthorizationID']
+                error = random.random() > ratio
+                status = 'Error' if error else 'Accepted'
+                self.stdout.write('Marking authorization {aid} for profile {cid} as {status}'.format(
+                    aid=aid,
+                    cid=cid,
+                    status=status,
+                ))
+                writer.writerow({
+                    'ClientAuthorizationID': aid,
+                    'ClientCandidateID': cid,
+                    'Status': status,
+                    'Date': now.strftime(PEARSON_DATETIME_FORMAT),
+                    'Message': 'Invalid ExamSeriesCode' if error else '',
+                })
+
+        self.write_zip(sftp_conn, result_file.getvalue(), now.strftime('eac-%Y-%m-%d.dat'), now)
+
+    def handle_cdd(self, sftp_conn, remote_path, ratio):
+        """Handle a CDD file"""
+        now = datetime.now(pytz.utc)
+        result_file = io.StringIO()
+        writer = csv.DictWriter(result_file, [
+            'ClientCandidateID',
+            'Status',
+            'Date',
+            'Message',
+        ], delimiter='\t')
+        writer.writeheader()
+        with sftp_conn.open(remote_path, mode='r') as cdd_file:
+            for row in csv.DictReader(cdd_file, delimiter='\t'):
+                cid = row['ClientCandidateID']
+                error = random.random() > ratio
+                status = 'Error' if error else 'Accepted'
+                self.stdout.write('Marking profile {cid} as {status}'.format(
+                    cid=cid,
+                    status=status,
+                ))
+                writer.writerow({
+                    'ClientCandidateID': cid,
+                    'Status': 'Error' if error else 'Accepted',
+                    'Date': now.strftime(PEARSON_DATETIME_FORMAT),
+                    'Message': 'Invalid Address' if error else '',
+                })
+
+        self.write_zip(sftp_conn, result_file.getvalue(), now.strftime('vcdc-%Y-%m-%d.dat'), now)
+
+    def write_zip(self, sftp_conn, data, filename, now):  # pylint: disable=no-self-use
+        """Write a zip file to the sftp server"""
+        zip_path = now.strftime('/tmp/ORGNAME-NS-%Y-%m-%d-%H%M%S.zip')
+        with zipfile.ZipFile(zip_path, 'w') as zf:
```

Comment:
This function is uncovered, just FYI. Feel free to skip if you don't think it's worth testing